### PR TITLE
remote: fix podman-remote play kube --userns

### PR DESCRIPTION
**Pull request by sstosh**
Fix `podman play kube --userns` to work in remote environment.

Related: #17392

#### Does this PR introduce a user-facing change?

```release-note
The remote Podman client’s `podman play kube` command now work --userns option.
```

### CODE REVIEW

**Comment by sstosh** [comment]
podman-remote --userns=auto works in rootless remote.

```console
$ bin/podman unshare cat /proc/self/uid_map
         0       1000          1
         1     524288      65536

$ bin/podman-remote play kube test.yaml
Pod:
d397dc09e978c0ed2416b0be6f8b7f2b517ca90b1ebc0742fe4c57c9d7710d3a
Container:
ef19b78add7b2696efecd79f4aed6f00ad71ce29a83166cfd62ef106de6c387a

$ bin/podman-remote exec testPod-testCtr cat /proc/self/uid_map
         0       1000          1
         1     524288      65536

$ bin/podman-remote play kube --replace --userns=auto test.yaml
Pods stopped:
d397dc09e978c0ed2416b0be6f8b7f2b517ca90b1ebc0742fe4c57c9d7710d3a
Pods removed:
d397dc09e978c0ed2416b0be6f8b7f2b517ca90b1ebc0742fe4c57c9d7710d3a
Secrets removed:
Volumes removed:
Pod:
1602ef3a9ebedbbcc3707cb9f406b1e5d9cda56fc77fa3f0cddef5eb83f77cea
Container:
0a17f031fe309d65a6a0ef4fa554fde54976ae7aa8e1db955b7fbf15357b14bb

$ bin/podman-remote exec testPod-testCtr cat /proc/self/uid_map
         0          1       1024
```

**Review by Luap99 — COMMENTED** [review]
Thanks, you also have to add the parameter to our swagger docs, see pkg/api/server/register_kube.go

**Review by vrothberg — COMMENTED** [review]
LGTM once the previous comment is addressed

**Review by Luap99 — APPROVED** [review]
/lgtm
/hold

**Comment by sstosh** [comment]
Sorry, I fixed the e2e test and re-pushed.

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

**Comment by fsdrw08** [comment]
may I know which version of the podman contain this commit will be release?

**Comment by rhatdan** [comment]
Looks like podman 4.7 in October.

**Pull request merged**
